fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/AvailableFields.java

```diff
@@ -109,17 +109,92 @@ public IndexKeyValueToPartialRecord.Builder buildIndexKeyValueToPartialRecord(@N
     /**
      * Get the fields that are available from just the index scan part of an index plan.
      */
-    public static AvailableFields fromIndex(@Nonnull RecordType recordType,
-                                            @Nonnull Index index,
-                                            @Nonnull PlannableIndexTypes indexTypes,
-                                            @Nullable KeyExpression commonPrimaryKey,
-                                            @Nonnull RecordQueryPlanWithIndex indexPlan) {
+    public static AvailableFields fromIndex(@Nonnull final RecordType recordType,
```

Comment:
This file mostly contains refactorings to allow the covering index code path for Lucene to directly translate the primary key information.